codice/ddf
# DDF-4082 Add relational validators to the ingest editor

#### What does this PR do?
Add relational validators to the ingest editor.
#### Who is reviewing it? 
@SmithJosh 
@jrnorth 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.
@vina
@andrewkfiedler 

#### How should this be tested?
Use the definitions files to create relational constraints. Add those attributes to the ingest editor configuration in intrigue. Try to upload files with attributes that violate those constraints and verify that the correct messages are displayed. Also configure some attribute aliases and verify that the messages use the aliased terms.

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4082](https://codice.atlassian.net/browse/DDF-4082)
#### Screenshots
![image](https://user-images.githubusercontent.com/8660335/44493245-1ec01500-a61c-11e8-9c26-3ee3b532e01d.png)


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
